### PR TITLE
ipa-cacert-manage prune: remove all expired certs

### PR DESCRIPTION
**Pull request by flo-renaud**
ipa-cacert-manage prune is removing the expired certs one at a time and this may result in verifying that one of the expired certs is still valid.
As a consequence, ipa-cacert-manage prune always fails when more than 1 cert are expired.

To avoid the issue, remove all the expired certs in a single pass, and validate only the ones that would remain after full pruning.

Fixes: https://pagure.io/freeipa/issue/9244
Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>

### CODE REVIEW

**Comment by flo-renaud** [comment]
The prune command is available from IPA 4.10+, a backport will be needed for ipa-4-10 only.

File: ipaserver/install/ipa_cacert_manage.py [hunk 1]
```diff
@@ -558,7 +560,8 @@ def prune(self):
         for ca_cert, ca_nickname, _ca_trust_flags in ca_certs:
             if ca_cert.not_valid_after < now:
                 expired_certs.append(ca_nickname)
-                self._delete_by_nickname(ca_nickname, self.options)
+
+        self._delete_by_nickname(expired_certs, self.options)
```
Comment (rcritten) [inline ipaserver/install/ipa_cacert_manage.py:564]: Can you add a ```if expired_certs:``` before this? It looks like _delete_by_nickname() will do a bunch of work even if there is nothing to prune:
- pull the list of certs
- create a temporary NSS db
- etc.

Comment (flo-renaud) [reply ipaserver/install/ipa_cacert_manage.py]: Good catch, thanks.

**Comment by rcritten** [comment]
Thanks for the patch. This looks good. ack when CI completes.

**Comment by flo-renaud** [comment]
CI is green, ack based on rcritten's comment

**Comment by flo-renaud** [comment]
master:

* 7aeb9e58600854045c2790550e3829df2c8e917d ipa-cacert-manage prune: remove all expired certs

**Pull request closed without merging**